File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/net/NativeConnectivityListener.java
package org.maplibre.android.net;

import androidx.annotation.Keep;

import org.maplibre.android.LibraryLoader;

/**
 * Updates the native library's connectivity state
 */
class NativeConnectivityListener implements ConnectivityListener {

  static {
    LibraryLoader.load();
  }

  @Keep
  private long nativePtr;
  @Keep
  private boolean invalidated;

  @Keep
  NativeConnectivityListener(long nativePtr) {
    this.nativePtr = nativePtr;
  }

  NativeConnectivityListener() {
    initialize();
  }

  @Override
  public void onNetworkStateChanged(boolean connected) {
    nativeOnConnectivityStateChanged(connected);
  }

  @Keep
  protected native void nativeOnConnectivityStateChanged(boolean connected);

  @Keep
  protected native void initialize();

  @Override
  @Keep
  protected native void finalize() throws Throwable;
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/net/NativeConnectivityListener.kt
package org.maplibre.android.net

import androidx.annotation.Keep
import org.maplibre.android.LibraryLoader

/**
 * Updates the native library's connectivity state
 */
internal class NativeConnectivityListener
    @Keep
    constructor(
        @field:Keep private var nativePtr: Long,
    ) : ConnectivityListener {
        @Keep
        private var invalidated = false

        constructor() : this(0) {
            initialize()
        }

        override fun onNetworkStateChanged(connected: Boolean) {
            nativeOnConnectivityStateChanged(connected)
        }

        @Keep
        protected external fun nativeOnConnectivityStateChanged(connected: Boolean)

        @Keep
        protected external fun initialize()

        @Keep
        @Throws(Throwable::class)
        protected external fun finalize()

        companion object {
            init {
                LibraryLoader.load()
            }
        }
    }
